### PR TITLE
CB-11245 Cloud provider information added to CDPOperationDetails as t…

### PR DESCRIPTION
**Pull request by nadamdb**
…his way this iformation is also included in Datalake and Datahub related messages, not just Environment requested messages.

See detailed description in the commit message.

### CODE REVIEW

File: structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/converter/CDPStructuredFlowEventToCDPOperationDetailsConverter.java [hunk 1]
```diff
@@ -36,6 +37,14 @@
             cdpOperationDetails.setCorrelationId(defaultIfEmpty(structuredOperationDetails.getUuid(), ""));
         }
 
+        EnvironmentDetails environmentDetails = cdpStructuredFlowEvent.getPayload();
+        if (environmentDetails != null) {
+            if (environmentDetails.getCloudPlatform() != null) {
```
Comment (topolyai5) [inline structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/converter/CDPStructuredFlowEventToCDPOperationDetailsConverter.java:42]: you can merge these ifs into a one

File: structuredevent-service-legacy/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/converter/StructuredFlowEventToCDPOperationDetailsConverter.java [hunk 2]
```diff
@@ -36,6 +37,14 @@
             cdpOperationDetails.setCorrelationId(defaultIfEmpty(structuredOperationDetails.getUuid(), ""));
         }
 
+        StackDetails stackDetails = structuredFlowEvent.getStack();
+        if (stackDetails != null) {
+            if (stackDetails.getCloudPlatform() != null) {
```
Comment (topolyai5) [inline structuredevent-service-legacy/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/converter/StructuredFlowEventToCDPOperationDetailsConverter.java:42]: same

**Review by topolyai5 — CHANGES_REQUESTED** [review]

File: usage-collection/src/main/proto/usage.proto [hunk 3]
```diff
@@ -1058,6 +1058,8 @@ message CDPOperationDetails {
     string flowState = 9;
     // Correlation ID to correlate events written together
     string correlationId = 10;
+    // Cloud provider identifier
+    CDPEnvironmentsEnvironmentType.Value environmentType = 11;
```
Comment (doktoric) [inline usage-collection/src/main/proto/usage.proto:1062]: the type is the provider info here?

Comment (nadamdb) [reply usage-collection/src/main/proto/usage.proto:1062]: yes, it's the same field as in CDPEnvironmentDetails

File: structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/converter/CDPStructuredFlowEventToCDPOperationDetailsConverter.java (hunk 1 above)
Comment (nadamdb) [reply structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/telemetry/converter/CDPStructuredFlowEventToCDPOperationDetailsConverter.java]: thanks, fixed it

**Review by topolyai5 — APPROVED** [review]

**Pull request merged**